src/Microsoft.DotNet.Build.CloudTestTasks/FilterForUpload.cs

```diff
@@ -0,0 +1,91 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using System.Runtime.InteropServices;
+using System.Text;
+
+using Microsoft.Build.Framework;
+using Microsoft.Build.Utilities;
+
+namespace Microsoft.DotNet.Build.CloudTestTasks
+{
+    public sealed class FilterForUpload : Task
+    {
+        //This task is used to select the appropriate test zips from the different output directories produced by the build,
+        //which are created based on OSGroup and TestTFM.
+
+        [Required]
+        public string TestOutputDirListsLocation { get; set; }
+
+        //This is the TestTFM folder that we want to select test zips from. This needs to be on the path to the test zip.
+        [Required]
+        public string FilterToTestTFM { get; set; }
+
+        //This is the OSGroup folder that we want to select test zips from. This does not require an exact match as some tests 
+        //may or may not have an exact OSGroup folder match.
+        [Required]
+        public string FilterToOSGroup { get; set; }
+        //This is the path under a given Platform/OSGroup/Configuration folder where the Test Archives are placed
+        [Required]
+        public string TestArchivesRoot { get; set; }
+
+        [Output]
+        public ITaskItem[] FilteredUploadItems
+        {
+            get { return this._copyFilteredUploadItems.ToArray(); }
+        }
+
+        private readonly List<ITaskItem> _copyFilteredUploadItems = new List<ITaskItem>();
+
+        public override bool Execute()
+        {
+            string[] filesToSearch = Directory.GetFiles(TestOutputDirListsLocation);
+            
+            foreach (var filename in filesToSearch)
+            {
+                string selectedTest = string.Empty;
+                using (StreamReader sr = new StreamReader(File.OpenRead(filename)))
+                {
+                    string line;
+                    while ((line = sr.ReadLine()) != null)
+                    {
+                        //Each line is of the format "$(BinDir)/tests/AnyOS.AnyCPU.Debug/Tests/netcoreapp1.0"
```

Comment:
Is this _literally_ "$(BinDir)" or is the expanded value?  If's the later, we'll have big problems, because my "BinDir" directory might contain all sorts of text you'll match on.